crates/sui-core/src/consensus_handler.rs

```diff
@@ -104,11 +104,10 @@ impl<T: ParentSync + Send + Sync> ExecutionState for ConsensusHandler<T> {
 
         /* (serialized, transaction, output_cert) */
         let mut transactions = vec![];
+        let timestamp = consensus_output.sub_dag.leader.header.created_at;
```

Comment:
Add a comment about why this must be `header.created_at`?

Reply:
If no test breaks here then it might also be a good opportunity to add some testing around it ? (even in a follow up PR). It seems quite sensitive part and someone can easily swap/change.